components/wal/wal_disk.go

```diff
@@ -1,53 +1,131 @@
 package wal
 
 import (
+	"encoding/json"
+	"time"
+
+	"github.com/aarthikrao/timeMachine/components/jobstore"
 	"github.com/aarthikrao/wal"
+	"github.com/pkg/errors"
 	"go.uber.org/zap"
+
+	jm "github.com/aarthikrao/timeMachine/models/jobmodels"
+)
+
+// logCommand specifies the type of operation for the wal command
+type logCommand byte
+
+var (
+	SetLog    logCommand = 0x01
+	DeleteLog logCommand = 0x02
 )
 
-type diskWAL struct {
+// logEntry is the wal log entry
+type logEntry struct {
+	Data       []byte     `json:"data,omitempty"`
+	Collection string     `json:"col,omitempty"`
+	Operation  logCommand `json:"op,omitempty"`
+}
+
+type walMiddleware struct {
 	w *wal.WriteAheadLog
+
+	next jobstore.JobStore
 }
 
+// Compile time interface check
+var _ jobstore.JobStoreConn = &walMiddleware{}
+var _ WALReader = &walMiddleware{}
+
 // InitaliseWriteAheadLog returns a instance of WAL on disk
 func InitaliseWriteAheadLog(
 	walDir string,
 	maxLogSize int64,
 	maxSegments int,
 	log *zap.Logger,
-) (*diskWAL, error) {
+	next jobstore.JobStore,
+) (*walMiddleware, error) {
 	w, err := wal.NewWriteAheadLog(&wal.WALOptions{
-		LogDir:      walDir,
-		MaxLogSize:  maxLogSize,
-		MaxSegments: maxSegments,
-		Log:         log,
+		LogDir:            walDir,
+		MaxLogSize:        maxLogSize,
+		MaxSegments:       maxSegments,
+		Log:               log,
+		MaxWaitBeforeSync: 1 * time.Second, // TODO: change this variable to default
```

Comment:
Sync once per second looks too much, we should have around 30 seconds. Ideally we should pass this as config

Reply:
Yes. We will have config for the same. 
Even with 1 sec, the OS will keep syncing in the background without us explicitly calling it.